### PR TITLE
Manual changelog descriptions

### PR DESCRIPTION
**Pull request by eike-hass**
# Description of change
Enable release descriptions through Markdown files.
The files live in `.github/releases` and must be named with the tag the same as the tag it is describing.

## Type of change
Add an `x` to the boxes that are relevant to your changes.

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [X] Enhancement (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Fix

## How the change has been tested
Describe the tests that you ran to verify your changes.
Make sure to provide instructions for the maintainer as well as any relevant configurations.

## Change checklist
Add an `x` to the boxes that are relevant to your changes.

- [X] I have followed the contribution guidelines for this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: .github/releases/v0.5.0-dev.1.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Hello World
+multi
```
Comment (cycraig) [inline .github/releases/v0.5.0-dev.1.md:2]:
```suggestion
This release introduces multiple breaking changes to the structure of IOTA DID Documents and their Tangle messages, rendering any identity created with a prior version incompatible and unresolvable. A versioning system has been introduced so any new identities should hopefully be forward compatible with any future breaking changes to the message structure.
```

File: .github/releases/wasm-v0.5.0-dev.1.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+Hello World
+multi
```
Comment (cycraig) [inline .github/releases/wasm-v0.5.0-dev.1.md:2]:
```suggestion
This release introduces multiple breaking changes to the structure of IOTA DID Documents and their Tangle messages, rendering any identity created with a prior version incompatible and unresolvable. A versioning system has been introduced so any new identities should hopefully be forward compatible with any future breaking changes to the message structure.
```

File: .github/actions/release/changelog-generator/action.yml [hunk 3]
```diff
@@ -45,6 +45,11 @@ runs:
           --future-release ${{ inputs.future-release }} 
           ${{ inputs.optional-arg }}
 
+    - name: Enrich changelog
```
Comment (cycraig) [inline .github/actions/release/changelog-generator/action.yml:48]: Opinion: maybe `changelog-descriptions` might be a more descriptive name for the script than `enrich-changelog`, or `populate-changelog-descriptions`?

```suggestion
    - name: Populate changelog descriptions
```

File: .github/releases/v0.3.0.md [hunk 4]
```diff
@@ -0,0 +1,2 @@
+Hello World 0.0.3
+multi
```
Comment (cycraig) [inline .github/releases/v0.3.0.md:2]:
```suggestion
This release introduces the high-level `Account` API for creating and managing IOTA identities.
```

File: .github/workflows/scripts/enrich-changelog.sh [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    echo "Processing $file"
+    filename=$(basename -- "$file")
+    filename="${filename%.*}"
+    # read the file contents
+    filecontent=$(cat $file)
+    # replace new lines in file contents, so we can pipe it into sed
+    replaced_content=$(echo $filecontent | sed '$!s/$/\\n/' | tr -d '\n')
+    # add file content after the line with the full changelog link (e.g. [Full Changelog](https://github.com/iotaledger/identity.rs/compare/v0.3.0...v0.5.0-dev.1) of corresponding version 
+    # note we use '@' as a delimiter because the content contains '\'
+    sed -i "\@\.\.\.$filename@a\ \n$replaced_content" $CHANGELOG_PATH
```
Comment (cycraig) [inline .github/workflows/scripts/enrich-changelog.sh:21]: Only thing I noticed is that this processes and removes escape characters.

E.g. In `v0.5.0-dev.1.md`
```
asdf \\ \t
```
becomes
```
asdf \	
```
in `CHANGELOG.md`.

**Review by cycraig — APPROVED** [review]
Looks great! I really appreciate being able to write changelog descriptions easily.

The escape characters being processed by sed is a minor concern but shouldn't break anything, hopefully? The script is not idempotent but that shouldn't be a problem since the changelogs are regenerated each time.

Aside: is it possible to alter the changelog template at all? There are minor changes I would like such as using "###" for the "Changed", "Added", "Patch" sub-headings and surrounding PR links with square brackets. Completely irrelevant to this PR however.

File: .github/workflows/scripts/enrich-changelog.sh (hunk 5 above)
Comment (eike-hass) [reply .github/workflows/scripts/enrich-changelog.sh:21]: we think it is `sed` doing that?

**Comment by eike-hass** [comment]
> Looks great! I really appreciate being able to write changelog descriptions easily.
> 
> The escape characters being processed by sed is a minor concern but shouldn't break anything, hopefully? The script is not idempotent but that shouldn't be a problem since the changelogs are regenerated each time.
> 
> Aside: is it possible to alter the changelog template at all? There are minor changes I would like such as using "###" for the "Changed", "Added", "Patch" sub-headings and surrounding PR links with square brackets. Completely irrelevant to this PR however.

changing the heading style could work through `configure-sections` / `add-sections`  (https://github.com/github-changelog-generator/github-changelog-generator/wiki/Advanced-change-log-generation-examples), but for more advanced templating we are out of luck I guess.

File: .github/workflows/scripts/enrich-changelog.sh (hunk 5 above)
Comment (cycraig) [reply .github/workflows/scripts/enrich-changelog.sh]: :shrug:

**Pull request merged**